openslides/users/serializers.py

```diff
@@ -57,19 +57,34 @@ class Meta:
 
     def validate(self, data):
         """
-        Checks that first_name or last_name is given. Generates the
+        Checks if the given data is empty. Generates the
         username if it is empty.
         """
-        if not (data.get('username') or data.get('first_name') or data.get('last_name')):
-            raise ValidationError({'detail': _('Username, given name and surname can not all be empty.')})
 
-        # Generate username. But only if it is not set and the serializer is not
-        # called in a PATCH context (partial_update).
         try:
             action = self.context['view'].action
         except (KeyError, AttributeError):
             action = None
 
+        # Check if one of the fields is actually filled when doing a PATCH
+        if action == 'partial_update':
+            filled_field_keys = []
+            for field in data:
+                if data.get(field) != '':
+                    filled_field_keys.append(field)
+
+            # Check if filled fields match to any of the given fields
+            intersecting_keys = list(set(filled_field_keys) & set(self.fields.keys()))
+            if len(intersecting_keys) < 1:
+                raise ValidationError({'detail': _('You cannot send an empty instruction')})
```

Comment:
Why is this a problem? The case that someone sends no valid fields is a standard case and is (I think so) already solved by the DjangoRESTFramework. I think what you only want to change is the case in the else clause. Right?

Reply:
The current issue is it is not possible to send a patch to a defined user ID, without either re-distributing the username, first or last name.

This could lead to extra steps in working with the rest frame work:
 * GET the Infos for the user ID first
 * Parse them
 * extract one of those: username, first or last name
 * build patch request
 * send PATCH

instead of just
 * build the patch request
 * send PATCH

Since there is objectively no need to redistribute one of the information during a patch, because an existing data object is changed, one should not be required to redeliver those, especially since those three fields are worthy of protection, since it can prevent the user from logging in.

On the issue of checking simply if one field is filled: It is just a convinience thing, since the class already holds a list of fields, I thought it would be more readable to just check if one is filled, that a change of data is truly happening.
Unfortunately the DjangoRESTFFramework does not prevent in sending an empty data set within a patch, it just won't set anything without a rejection. To prevent unnecessary calculations or workflow triggering this simply check would be sufficient, I think.

Reply:
Done!